build_tools/packaging/python/generate_release_index.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
+"""Creates an HTML page for releases for `pip install --find-links` from a subdirectory in an S3 bucket.
+
+Sample usage:
+
+    ```bash
+    ./build_tools/packaging/python/generate_release_index.py \
+        --bucket=therock-dev-python \
+        --endpoint=s3.us-east-2.amazonaws.com \
+        --subdir=gfx110X-dgpu \
+        --output=index.html
+    ```
+"""
+
+import argparse
+import boto3
+import html
+import io
+import sys
+import textwrap
+
+
+def parse_arguments():
+    p = argparse.ArgumentParser()
+    p.add_argument(
+        "--bucket",
+        required=True,
+        help="S3 bucket name",
+    )
+    p.add_argument(
+        "--endpoint",
+        default="s3.us-east-2.amazonaws.com",
+        help="S3 endpoint",
+    )
+    p.add_argument(
+        "--subdir",
+        "--subdirectory",
+        required=True,
+        help="Subdirectory in S3 bucket",
+    )
```

Comment:
Could add more information to these help strings. What does the script do with the bucket, endpoint, and subdir? If I want to _test_ this, what arguments do I need to change and how?